lib/auth/auth_with_roles.go

```diff
@@ -1924,9 +1924,15 @@ func (a *ServerWithRoles) NewKeepAliver(ctx context.Context) (types.KeepAliver,
 // user is allowed to assume the returned roles.
 func (a *ServerWithRoles) determineDesiredRolesAndTraits(req proto.UserCertsRequest, user types.User) ([]string, wrappers.Traits, error) {
 	if req.Username == a.context.User.GetName() {
+		// If RoleRequestsOnly is set, make sure we don't return unusable
+		// certs: an identity without roles can't be parsed.
+		if req.RoleRequestsOnly && len(req.RoleRequests) == 0 {
+			return nil, nil, trace.BadParameter("at least one role request is required")
+		}
```

Comment:
Do we want to add a future version date in which we will deprecate the ability to provide `req.RoleRequests` without the `RoleRequestsOnly` flag ?

Reply:
That's probably for the best. I've added a note and preemptively renamed the parameter to `UseRoleRequests` which I think will make more sense when it's required.